lib/twitter/rest/tweets.rb

```diff
@@ -300,6 +296,45 @@ def retweeters_ids(*args)
 
     private
 
+      # Uploads images and videos. Videos require multiple requests and uploads in chunks of 5 Megabytes.
+      # The only supported video format is mp4.
+      #
+      # @see https://dev.twitter.com/rest/public/uploading-media
+      def upload(media) # rubocop:disable MethodLength, AbcSize
+        if !(File.basename(media) =~ /\.mp4$/)
```

Comment:
I store my videos on S3 using refile, thus they don't have a `.mp4` basename, I guess checking for the basename to determine if something is a video isn't the best way to do it.